snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+
+        /snap/<snap name>/<snap revision>
+        """
+        return os.path.join('/', self.rel_prefix)
+
+    @property
+    def jhsource(self):
+        """Get the absolute path to the JHBuild source in the build container.
+
+        /snap/<snap name>/<snap revision>/usr/src/jhbuild
+        """
+        return os.path.join(self.prefix, 'usr', 'src', 'jhbuild')
+
+    @property
+    def rel_host(self):
+        """Get the relative path to the mount points in the build container.
+
+        host
+        """
+        return os.path.join('host')
+
+    @property
+    def rel_jhtarballs(self):
+        """Get the relative path to the JHBuild tarballs in the build container.
+
+        host/jhtarballs
+        """
+        return os.path.join(self.rel_host, 'jhtarballs')
+
+    @property
+    def rel_jhmirror(self):
+        """Get the relative path to the JHBuild DVCS mirror repositories in the build container.
+
+        host/jhmirror
+        """
+        return os.path.join(self.rel_host, 'jhmirror')
+
+    @property
+    def rel_jhcheckout(self):
+        """Get the relative path to the JHBuild DVCS checkout repositories in the build container.
+
+        host/jhcheckout
+        """
+        return os.path.join(self.rel_host, 'jhcheckout')
+
+    @property
+    def rel_ccache(self):
+        """Get the relative path to the ccache cache in the build container.
+
+        host/ccache
+        """
+        return os.path.join(self.rel_host, 'ccache')
+
+    @property
+    def jhtarballs(self):
+        """Get the absolute path to the JHBuild tarballs in the build container.
+
+        /host/jhtarballs
+        """
+        return os.path.join('/', self.rel_jhtarballs)
+
+    @property
+    def jhmirror(self):
+        """Get the absolute path to the JHBuild DVCS mirror repositories in the build container.
+
+        /host/jhmirror
+        """
+        return os.path.join('/', self.rel_jhmirror)
+
+    @property
+    def jhcheckout(self):
+        """Get the absolute path to the JHBuild DVCS checkout repositories in the build container.
+
+        /host/jhcheckout
+        """
+        return os.path.join('/', self.rel_jhcheckout)
+
+    @property
+    def ccache(self):
+        """Get the absolute path to the ccache cache in the build container.
+
+        /host/ccache
+        """
+        return os.path.join('/', self.rel_ccache)
+
+    @property
+    def jhbuildrc(self):
+        """Get the absolute path to the JHBuild configuration in the build container.
+
+        /snap/<snap name>/<snap revision>/etc/jhbuildrc
+        """
+        return os.path.join(self.prefix, 'etc', 'jhbuildrc')
+
+    # TODO: use custom /etc/fonts/fonts.conf
+    @property
+    def xdgdatahome(self):
+        """Get the absolute path to XDG_DATA_HOME in the build container.
+
+        /snap/<snap name>/<snap revision>/usr/share
+        """
+        return os.path.join(self.prefix, 'usr', 'share')
```

Comment:
We usually define the attributes in the init, not as properties. Because then the statement is executed only once, not every time the attribute is accessed.